Adapter.H

```diff
@@ -177,68 +180,67 @@ private:
     //- Checkpointed volScalarField mesh fields (copies)
     std::vector<Foam::volScalarField::Internal*> volScalarInternalFieldCopies_;
 
-    // Vectors of pointers to the checkpointed fields and their copies
+    // Maps of pointers to checkpointed fields and their copies
```

Comment:
As discussed, this might not be necessary.
Let's try a quick implementation using `std::vector` and, if this gets uglier, then a map should be ok as well.